pkg/cluster/spec/util.go

```diff
@@ -57,7 +57,7 @@ func ClusterMetadata(clusterName string) (*ClusterMeta, error) {
 	var cm ClusterMeta
 	err := GetSpecManager().Metadata(clusterName, &cm)
 	if err != nil {
-		return nil, errors.AddStack(err)
+		return &cm, errors.AddStack(err)
```

Comment:
Should we return the `cm` even though error encountered?

Reply:
Yes, it is possible the error could be ignored and the `cm` continue be processed. A typical scenario is when the topology file is broken (from `edit-config`, manual modification, or a broken legacy release), we could keep a max possibility that operations like `display` and `destroy` are still working.